pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -186,8 +187,8 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration, featur
 			allErrors = append(allErrors, fmt.Errorf("invalid configuration: shutdownGracePeriodCriticalPods %v must be either zero or otherwise >= 1 sec", kc.ShutdownGracePeriodCriticalPods))
 		}
 	}
-	if (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) && !localFeatureGate.Enabled(features.GracefulNodeShutdown) {
-		allErrors = append(allErrors, fmt.Errorf("invalid configuration: specifying shutdownGracePeriod or shutdownGracePeriodCriticalPods requires feature gate GracefulNodeShutdown"))
+	if (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) && !gracefulNodeShutdownEnabled {
+		allErrors = append(allErrors, fmt.Errorf("invalid configuration: specifying shutdownGracePeriod or shutdownGracePeriodCriticalPods requires feature gate GracefulNodeShutdown or WindowsGracefulNodeShutdown"))
```

Comment:
I think we should require feature gate GracefulNodeShutdown on all node OSs, and also check for a gate specific to that node OS.

Ideally we'd have done that before shipping OS-dependent support, but it's too late now. We can still fix it ahead of graduation, and before the Windows support goes on by default.

People may legitimately want to turn off only one part of this (eg just the Linux support, keep the beta Windows support enabled).